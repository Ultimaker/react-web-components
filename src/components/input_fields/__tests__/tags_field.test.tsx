// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import TagsField, { TagsFieldProps } from '../tags_field';
import InputFieldWrapper from '../input_field_wrapper';
import TagsSelector from '../../tags_selector';

describe('The tags field component', () => {
    let props: TagsFieldProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputField',
            onChangeHandler: jest.fn(),
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
            staticField: false,
            value: ['tag1', 'tag2'],
            placeholder: 'placeholder text',
            tagSuggestions: ['tag3', 'tag4'],
        };
        wrapper = shallow(<TagsField {...props} />);
    });

    it('should render a wrapper', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

    it('should render a tags selector', () => {
        expect(wrapper.find(TagsSelector).props()).toEqual({
            id: props.id,
            onChangeHandler: wrapper.instance()._onChange,
            value: props.value,
            placeholder: props.placeholder,
            suggestions: props.tagSuggestions,
            error: undefined,
            autofocus: false,
            disabled: false,
        });
    });

    it('should call the callback', () => {
        expect(props.onChangeHandler).not.toHaveBeenCalled();
        wrapper.find(TagsSelector).prop('onChangeHandler')(['tag5']);
        expect(props.onChangeHandler).toHaveBeenCalledWith(props.id, ['tag5']);
    });
});
