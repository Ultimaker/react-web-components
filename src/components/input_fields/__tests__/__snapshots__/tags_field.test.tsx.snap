// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The tags field component should render a wrapper 1`] = `
<InputFieldWrapper
  className="input-field--tags"
  id="testInputField"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  onChangeHandler={[MockFunction]}
  showValidationError={true}
  staticField={false}
  touched={false}
>
  <TagsSelector
    autofocus={false}
    disabled={false}
    id="testInputField"
    onChangeHandler={[Function]}
    placeholder="placeholder text"
    suggestions={
      Array [
        "tag3",
        "tag4",
      ]
    }
    value={
      Array [
        "tag1",
        "tag2",
      ]
    }
  />
</InputFieldWrapper>
`;
